devito/operator/operator.py

```diff
@@ -865,6 +865,11 @@ def _emit_timings(timings, indent=''):
 
     def _emit_apply_profiling(self, args):
         """Produce a performance summary of the profiled sections."""
+
+        # In case 'MPI0' is selected for logging, restrict result printing to one rank
+        if configuration['mpi']:
+            set_log_level(configuration['log-level'], comm=args.comm)
```

Comment:
I still don't really like changing configuration like that unless it's a temporary `swithconfig`

Reply:
Based on this comment: https://github.com/devitocodes/devito/pull/2130#discussion_r1197496681

I thought that we would be happy to happen here?

Reply:
no, this would change user-level setup after the first Operator run, so afterwards we would only see output from one rank